core/src/main/java/io/questdb/cairo/ColumnType.java

```diff
@@ -393,7 +458,15 @@ public static boolean isSymbolOrStringOrVarchar(int columnType) {
     }
 
     public static boolean isTimestamp(int columnType) {
-        return columnType == TIMESTAMP;
+        return ColumnType.tagOf(columnType) == TIMESTAMP;
+    }
+
+    public static boolean isTimestampMicro(int timestampType) {
+        return (timestampType & TIMESTAMP_TYPE_MASK) == TIMESTAMP_MICRO;
```

Comment:
Why is the AND mask needed here? On master we didn't use it.

Reply:
`getTimestampDriver()` method in this branch used the mask, so I ported it to other checks. AFAIU the reason is that the `timestampType` may have designated ts bit flag set on it.

Reply:
Yes, that would be my idea, too. However, we don't mask out the designated ts bit on master, and I don't think anything about that has changed in this branch.

Reply:
One of the reasons we may have seen the need for this goes again back to my concern about the placement of the nanos type bit. If that bit was inside the short, we could use `switch (tagOf(type))`, which automatically removes the designated ts bit, but would keep the nanos type bit. This way, we must switch on `int` and be worried whether it has the designated bit.